File: Contest/LeetCodeWeeklyContest/WeeklyContest248/3_fail2.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        """
        even = 0, 2, 4, 6, 8
        prime = 2, 3, 5, 7
        """
        MOD = 10 ** 9 + 7
        ans = 1
        ans *= 5 ** (n - n // 2)
        ans *= 4 ** (n // 2)
        ans %= MOD
        return ans

# TLE
# 806166225460393

File: Contest/LeetCodeWeeklyContest/WeeklyContest248/3_fail2.cpp
#include <math.h>
class Solution
{
public:
    int countGoodNumbers(long long n)
    {
        long long MOD = pow(10, 9) + 7;
        long long ans = 1;
        ans *= (long long)pow(5, n - n / 2) % MOD;
        ans %= MOD;
        ans *= (long long)pow(4, n / 2) % MOD;
        ans %= MOD;
        return ans;
    };
};

// WA
// 50
